---
title: "Help page"
author: "GREIN"
date: "2024-11-10"
output: openintro::lab_report
---

# R MarkDown常用语法介绍

## Markdown 标题语法

创建标题，在标题前添加井号 (#) 。# 的数量代表了标题的级别，注意#后需要添加一个空格生效。

示例：

```
# 一级标题
## 二级标题
### 三级标题
......
###### 六级标题
```

![标题语法](https://files.mdnice.com/user/71304/29563a1d-8ef0-4118-8185-481a05754315.png)

## Markdown 段落语法

要创建段落，请使用空白行将一行或多行文本进行分隔。

示例：

```
R Markdown是一种用于数据科学写作的框架，它的核心思想是创作包含文本和代码的动态文档。

R Markdown可以用于撰写数据分析报告、记录学习过程、发表电子书和长篇文档。
```

## Markdown 换行语法

在一行的末尾添加两个或多个空格，然后按回车键，即可创建一个换行。

示例：

```
R Markdown是一种用于数据科学写作的框架，  
它的核心思想是创作包含文本和代码的动态文档。
```

## Markdown 强调语法

- **字体加粗**

要加粗文本，需要再内容前两个星号（ ** ）或下划线（ __ ）。如需加粗一段文字的中间部分，则需要在加粗部分的两侧各添加两个星号（ ** ）。

- *斜体*

要用斜体显示文本，请在文本前后添加一个星号（ * ）或下划线（ _ ）。要斜体突出文本的中间部分，请在文字前后各添加一个星号，中间不要带空格。

- ***粗体和斜体***

要同时用粗体和斜体突出显示文本，请在文段的前后各添加三个星号（ *** ）或下划线（ ___ ）。要加粗并用斜体显示文本的中间部分，请在要突出显示的部分前后各添加三个星号，中间不要带空格。

示例：

```
**字体加粗**

*斜体*

***粗体和斜体***
```

## Markdown 列表语法

- 有序列表

创建一个有序列表，需要在每一条内容前增加一个数字和英文句号. 列表的起始数据必须是1：

示例：

```
1. 生物信息学
2. biomamba
3. 生信基地
4. MarkDown
5. Shiny
或者：
1. 生物信息学
1. biomamba
1. 生信基地
1. MarkDown
1. Shiny
```

1. 生物信息学
2. biomamba
3. 生信基地
4. MarkDown
5. Shiny

或者：

1. 生物信息学
1. biomamba
1. 生信基地
1. MarkDown
1. Shiny

- 无序列表

创建一个无序列表，需要在每一条内容前增加破折号（-）、星号（ * ）或加号（+）。缩进一个或多个列表项可创建嵌套列表。

示例：

```
- 生物信息学
- biomamba
- 生信基地
或者：
* 生物信息学
* biomamba
* 生信基地
或者：
+ 生物信息学
+ biomamba
+ 生信基地

- 生物信息学
 - biomamba
  - 生信基地
```

- 生物信息学
- biomamba
- 生信基地

或者：

* 生物信息学
* biomamba
* 生信基地

或者：

+ 生物信息学
+ biomamba
+ 生信基地

- 生物信息学
  - biomamba
    - 生信基地
    
## Markdown 代码语法

通常代码模块，可以使用一个或多个反引号：

示例：

![](https://files.mdnice.com/user/71304/980bb82b-a409-47a9-80ee-b9c0756881dc.png)

```
library(shiny)
library(DT)
library(shinyjs)
library(shinycssloaders)
library(shinyWidgets)
library(shinydashboard)
library(markdown)
```

## Markdown 分隔线

要创建分隔线，在单独一行上使用三个或多个星号（ *** ）、破折号（ --- ）或下划线（ ___ ），并且不能包含其他内容。

```
***
---
___
```

***

---

___

## Markdown 链接语法

- 超链接Markdown语法代码：[超链接显示名](超链接地址)

示例：

```
这是一个链接 [Biomanba生信基地](https://space.bilibili.com/484326544/)
```

这是一个链接 [Biomanba生信基地](https://space.bilibili.com/484326544/)

- 网址和Email地址

使用尖括号可以很方便地把URL或者email地址变成可点击的链接。

```
<https://space.bilibili.com/484326544/>
```

<https://space.bilibili.com/484326544/>

## Markdown 图片语法

要添加图像，请使用感叹号（!）, 然后在方括号增加替代文本，图片链接放在圆括号里，括号里的链接后可以增加一个可选的图片标题文本。

示例：

```
![Biomamba 生信基地](https://files.mdnice.com/user/71304/9a799584-11f8-4d76-a64a-86bf4f4ac117.jpg)
```

给图片增加链接，请将图像的Markdown括在方括号中，然后将链接添加在圆括号中。

```
[![Biomamba 生信基地](https://files.mdnice.com/user/71304/9a799584-11f8-4d76-a64a-86bf4f4ac117.jpg)](https://space.bilibili.com/484326544/)
```

渲染效果如下：

[![Biomamba 生信基地](https://files.mdnice.com/user/71304/9a799584-11f8-4d76-a64a-86bf4f4ac117.jpg)](https://space.bilibili.com/484326544/)


# Landing page (GEO datasets)

To illustrate the usability and efficacy of GREIN, we will walk through the available features for exploring and analyzing data sets with an example. The interpretation of the results need further bioinformatics expertise. GREIN is a platform independent web application. All you need is to open a web browser and type this url: https://shiny.ilincs.org/grein in the address bar which will load the following landing page with three sections as shown below:

![Figure 1: GREIN landing page.](https://files.mdnice.com/user/71304/2e251f8c-df26-430a-89d8-355a8405a555.png)

## Section 1

The first section provides information regarding the sample size distribution of the processed datasets, total number of data sets already processed or waiting to be processed, and the number of processed human, mouse, or rat samples by our GREP2 pipeline.

## Section 2

The first panel in section 2 provides the option to process your GEO dataset of interest if it is not already processed. You can search for a GEO series accession to see if it exists in the dataset table (Section 3). If not, then Start processing button will appear right below this box and you can initialize the process by clicking this button which will take you to the Processing console window (See Fig. 1). You will see your data set id in the Currently processing or at the bottom of the Waiting to process menu. This window also shows the logs of the currently processing dataset requested by a user. A single server processing pipeline is continuously running and processing datasets whenever requested. This pipeline is dedicated to process the user requested datasets only. Depending on the size of the data and queue, the requested data sets are automatically uploaded to the portal as soon as they are processed.

![Figure 2: Processing console window. Clicking the Show log file button will open the processing log of user requested dataset.](https://files.mdnice.com/user/71304/e924dc37-560f-495e-bae1-5aaaa16c548c.png)

You can also search by biomedical ontologies (for example, cancer, basal cell, kidney, etc.) in the second panel of this section. We use ontology terms mapped to GEO samples by MetaSRA project. Your search term associated ontologies can be found in the Metadata under Explore dataset tab. The final panel in this section shows the user requested data sets. If you have already requested for a data set to process, then you will be able to see your data set id (GEO series accession) once you refresh the web page. You can also see the status of the processing queue by pressing the Processing console button.

## Section 3

You can see the list of processed data sets with additional information in the data table shown in section 4 (See Fig. 1). Two types of search options are available in this table. Search box at the top-right of the table lets a user to search anything in this table. Other search boxes at the top of each column enables column-wise searching. You can click the GEO accession in the first column to start exploring a dataset.

# Explore dataset

Let us demonstrate the features of GREIN for exploring and analyzing an RNA-seq data by searching GSE60450 either at the top-right or first column’s search box in the dataset table. If you click GSE60450, it will take you to the Explore dataset tab. This experiment was conducted to examine the change in expression profiles between luminal and basal cells in mouse mammary glands of virgin, pregnant, and lactating mice. The data set is available in GEO as GSE60450.

## Description

This tab panel provides descriptive information including study link, number of GEO samples, number of SRA runs, title, and study summary of the corresponding dataset.

![Figure 3: Description tab panel. The study link will take the user to GEO web page.](https://files.mdnice.com/user/71304/71772913-c742-4532-85dc-748c6ad469e6.png)

## Metadata

GEO metadata contains a lot of information, although not all of these are useful for analysis or visualization purpose. So, we provide a filtered version of the metadata besides the full metadata.

![Figure 4: Metadata tab panel. Both the full and filtered metadata are downloadable in CSV format.](https://files.mdnice.com/user/71304/f1f4a10b-d2cd-4c4c-9753-5c46db47e5e4.png)

We filter metadata based on the following criteria:

* Columns that contain a single value.

* Columns with incoherent information regarding analysis and visualization such as dates, time, download path and so on.

This dataset (GSE60450) has two cell types and three developmental stages and each combination has two biological replicates. You can also download both the filtered and full metadata.

## Counts table

This table shows gene wise estimated read abundance (rounded to the nearest integer) for each sample. We use Salmon to quantify transcript abundances for each sample. These transcript level estimates are then summarized to gene level using Bioconductor package tximport which gives estimated counts scaled up to library size while taking into account for transcript length. We obtained gene annotation for Homo sapiens (GRCh38), Mus musculus (GRCm38), and Rattus norvegicus (Rnor_6.0) from Ensemble (release-91). Both gene and transcript level expression data are downloadable.

![Figure 5: Counts table tab panel. Both the gene and transcript level counts are downloadable in CSV format. You can change the number of samples or genes to show. A global search option is also available.](https://files.mdnice.com/user/71304/93cb757c-1b9b-4f21-ba99-edb0e824072e.png)

## QC report

After running FastQC and Salmon, we generate a combined quality control report of all the samples using MultiQC. This downloadable report contains information regarding read mapping and quality scores of the FastQ files. In the general statistics table, each sample corresponds to two rows, the first one for the Salmon read mapping and the second one for FastQC (See Fig. 6).

![Figure 6: MultiQC report. Both the FastQC and Salmon alignment reports are available for each of the samples. Besides the whole HTML report, all the tables and figures and individually downloadable.](https://files.mdnice.com/user/71304/966e4973-1c1e-423a-a9e5-7391c4c607a2.png)

## Visualization

This section provides access to four different types of interactive exploratory plots. These plots are important in order to uncover underlying relationship of the samples and gain deeper insight of the data structure. We leverage several state-of-the-art R and Bioconductor packages for this purpose.

## Correlation plot

Sample-wise correlation heatmap is generated using Plotly. User can hover over the heatmap to see Pearson correlation coefficient values or zoom in to any specific area and double click to zoom out.

![Figure 7: Correlation plot tab panel. User can select any number of samples.](https://files.mdnice.com/user/71304/33acbc0a-128f-418c-8e38-53517cd83947.png)

Here, overall sample-to-sample correlation is quite high, although basal or luminal cells have higher within group correlation compared to between group correlation.

